enhancements/baremetal-IPI/baremetal.md

```diff
@@ -0,0 +1,220 @@
+below.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+Baremetal IPI deployments enable OpenShift to enroll baremetal servers to become
+Machines that can be consumed to run K8s workloads.
+The Baremetal Operator (https://github.com/metal3-io/baremetal-operator) along
+with other provisioning services (Ironic and dependencies) run in their own
+pod called the "metal3". This pod is deployed by the Machine API Operator when
+the Platform type is Baremetal. The OpenShift Installer is responsble for
+providing all the necessary configs required for a successful deployment.
+
+## Motivation
+
+The motivation for this enhancement request is to provide a background for all the
+the subsequent enhancement requests for Baremetal IPI deployments.
+
+### Goals
+
+The goal of this enhancement request is to provide context for all the changes
+that have already been merged towards making Baremetal IPI deployments a reality.
+All future Baremetal enhancement requests will refer back to this one to provide
+context.
+
+### Non-Goals
+
+No development PRs will be raised as a result of this enhancement request.
+
+## Proposal
+
+Every OpenShift based Baremetal IPI deployment will run a "metal3" pod on
+all Master Nodes. A "metal3" pod includes a container running BareMetal
+Operator(BMO) and several other supporting containers that work togther.
+
+The BMO and other supporting containers together are able to discover a
+baremetal server in a pre-determined provisioning network, learn the
+HW attributes of the server and eventually boot it to make it available
+as a Machine within a MachineSet.
+
+The Machine API Operator currently deploys the "metal3" pod only when the
+Platform type is Baremetal. The MAO also installs the BaremetalHostCRD
+for every type of deployment starting from release 4.2.
+
+The installer is responsible for kicking off a Baremetal IPI deployment
+with the right configuration.
+
+A set of follow-up enhancement requests will try to bring in the following
+enhancements:
+1. Providing the BMO and it supporting containers with configuration that
+is specific to the server and its provisioning environment. This enhancement
+request is already under review here : 
+https://github.com/openshift/enhancements/pull/90
+
+2. The MAO would also be responsible for deploying a DeamonSet that would
+run on all baremetal servers to monitor the hardware inventory on each
+of these servers.
+
+### User Stories [optional]
```

Comment:
We should either remove this section, or remove the "(optional)" and add some stories like deploying the control plane, adding workers, and removing workers.

Reply:
I wonder id we shouldn't describe how a new node gets provisioned from start to finish in here somewhere to really explain the idea in full?

Reply:
The latest version has this information. Please let me know if everything has been captured accurately.